utils/switchable/snapshot_test.go

```diff
@@ -0,0 +1,140 @@
+package switchable
+
+import (
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/Fantom-foundation/lachesis-base/common/bigendian"
+	"github.com/Fantom-foundation/lachesis-base/kvdb"
+	"github.com/Fantom-foundation/lachesis-base/kvdb/memorydb"
+	"github.com/stretchr/testify/require"
+)
+
+func decodePair(b []byte) (uint32, uint32) {
+	v1 := bigendian.BytesToUint32(b[:4])
+	v2 := bigendian.BytesToUint32(b[4:])
+	return v1, v2
+}
+
+type UncallabaleAfterRelease struct {
```

Comment:
How about https://github.com/Fantom-foundation/go-opera/commit/b689c69292de6a45b32d9eba3ed0b300b99e2731 addition ?

Reply:
Yeah sure, but no need in RW mutex, as it mutates the slice in both bases